pkg/util/log/middleware.go

```diff
@@ -0,0 +1,145 @@
+package log
+
+import (
+	"bufio"
+	"fmt"
+	"net"
+	"net/http"
+	"strings"
+	"time"
+
+	kitlog "github.com/go-kit/kit/log"
+)
+
+func LoggingHandler(logger kitlog.Logger, next http.Handler) http.Handler {
+	return http.HandlerFunc(func(rw http.ResponseWriter, request *http.Request) {
+		wrapper := makeWrapper(rw)
+
+		host, _, err := net.SplitHostPort(request.RemoteAddr)
+		if err != nil {
+			host = request.RemoteAddr
+		}
+
+		privateLogger := kitlog.With(logger,
```

Comment:
The statement above about using contextual loggers sparingly applies here.

Reply:
fixed